beacon_node/network/src/beacon_processor/worker/gossip_methods.rs

```diff
@@ -1262,6 +1272,24 @@ impl<T: BeaconChainTypes> Worker<T> {
                 // Do not propagate these messages.
                 self.propagate_validation_result(message_id, peer_id, MessageAcceptance::Ignore);
             }
+            AttnError::PastSlot { .. } => {
+                // Produce a slot clock frozen at the time we received the message from the
+                // network.
+                let seen_clock = &self.chain.slot_clock.freeze_at(seen_timestamp);
+                let hindsight_verification =
+                    attestation_verification::verify_propagation_slot_range(
+                        seen_clock,
+                        failed_att.attestation(),
+                    );
+
+                // Only penalize the peer if it would have been invalid at the moment we received
+                // it.
+                if hindsight_verification.is_err() {
```

Comment:
In this case, if the hindsight_verification is an error, we could have just dropped the message

Reply:
I did consider adding an expired when the message comes into the queue. However, when the `BeaconProcessor` receives a message it either processes it or queues it. There aren't any existing cases where it does some processing (expired check) and then queues it.

I'm not fully against adding per-queueing processing, but I'm hesitant about it. If we add too much verification there then we can start to block the core tokio thread and other (potentially higher-priority) messages. I'm just presently not convinced that late attestations are causing us enough trouble to go and start modifying adding that complexity.